sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -559,6 +567,195 @@ def set_read_only(
 
+    @distributed_trace
+    def create_snapshot(
+        self,
+        name: str,
+        filters: List[Dict[str, str]],
+        *,
+        composition_type: Optional[str] = None,
+        retention_period: Optional[int] = None,
+        tags: Optional[Dict[str, str]] = None,
+        **kwargs
+    ) -> ConfigurationSettingSnapshot:
+        """Create a snapshot of the configuration settings.
+
+        :param name: The name of the snapshot to create.
+        :type name: str
+        :param filters: A list of filters used to filter the configuration settings by key field and label field
+            included in the snapshot. Each filter should be in format {"key": <value>, "label": <value>},
+            and key field value is required.
+        :type filters: list[dict[str, str]]
+        :keyword str composition_type: The composition type describes how the key-values within the
+            snapshot are composed. The 'all' composition type includes all key-values. The 'group_by_key'
+            composition type ensures there are no two key-values containing the same key. Known values are:
+            "all" and "group_by_key".
+        :keyword int retention_period: The amount of time, in seconds, that a snapshot will remain in the
+            archived state before expiring. This property is only writable during the creation of a
+            snapshot. If not specified, will set to 2592000(30 days). If specified, should be
+            in range 0 to 7776000(90 days). When set to 0, the snapshot will be deleted when archiving it.
+        :keyword dict[str, str] tags: The tags of the snapshot.
+        :return: The ConfigurationSettingSnapshot returned from the service.
+        :rtype: :class:`~azure.appconfiguration.models.ConfigurationSettingSnapshot`
+        :raises: :class:`HttpResponseError`, :class:`ClientAuthenticationError`, :class:`ResourceExistsError`
+        """
+        snapshot = ConfigurationSettingSnapshot(
+            filters=filters, composition_type=composition_type, retention_period=retention_period, tags=tags
+        )
+        error_map = {401: ClientAuthenticationError, 412: ResourceExistsError}
+        try:
+            response = self._impl.create_snapshot(
+                name=name,
+                entity=snapshot._to_generated(),
+                error_map=error_map,
+                **kwargs
+            )
+        except HttpResponseError as error:
+            raise error
+        return ConfigurationSettingSnapshot._from_generated(response)
+
+    @distributed_trace
+    def archive_snapshot(
+        self,
+        name: str,
+        *,
+        match_condition: MatchConditions = MatchConditions.Unconditionally,
+        etag: Optional[str] = None,
+        **kwargs
+    ) -> ConfigurationSettingSnapshot:
+        """Archive a configuration setting snapshot. It will update the status of a snapshot from "ready" to "archived".
```

Comment:
We could add more details here regarding what it means for the snapshot to be archived. For example, that after being archived, it will remain recoverable for the set `retention_period`, after which it will be deleted.